### DISCUSSION TITLE
Is it possible to get SVG with transparent background?

### DISCUSSION THREAD

**Discussion by xyz-coder** [ImageMagick/ImageMagick#3552] [Help]
I'm trying to load a SVG file and maintain its alpha channel/transparent background.

I've researched this quite a bit, and although there are hints that it should be possible, I've not seen anyone report that they have achieved it.

Here is my test code:

```
mwand := NewMagickWand();

// Set background to none
bgWand := NewPixelWand();
PixelSetColor( bgWand, 'none' );
MagickSetImageBackgroundColor( mwand, bgWand );

MagickReadImage(mwand, PAnsiChar( 'C:\lion-svgrepo-com.svg' ));

// Check image has alpha
isAlpha := MagickGetImageAlphaChannel( mwand );
if not isAlpha then
  exit;

width  := MagickGetImageWidth( mwand );
height := MagickGetImageHeight( mwand );

Allocate mem and get pixels
imagedata := allocMem( width*height*4 );
MagickExportImagePixels( mwand, 0, 0, width, height, PAnsiChar('RGBA'), imck_CharPixel, imagedata );

// Check output
ShowMessage( format( 'R:%d G:%d B:%d A:%d'#13#10'R:%d G:%d B:%d A:%d', [pbytearray(imagedata)^[0], pbytearray(imagedata)^[1], pbytearray(imagedata)^[2], pbytearray(imagedata)^[3], pbytearray(imagedata)^[4], pbytearray(imagedata)^[5], pbytearray(imagedata)^[6], pbytearray(imagedata)^[7] ] ));

DestroyMagickWand( mwand );
DestroyPixelWand( bgWand );
```

For all the images I test, such as:

https://www.svgrepo.com/svg/206727/lion

I always get Alpha of 255:

R:255 G:255 B:255 A:255
R:255 G:255 B:255 A:255

Is it possible to get the SVG alpha channel? Is there anything wrong with my code?

Thanks IA

**Comment by snibgo** [comment]
Instead of MagickSetImageBackgroundColor(), try MagickSetBackgroundColor().

**Reply by xyz-coder** [reply]
You are correct that MagickSetBackgroundColor() has an effect. Now PixelSetColor() changes the RGB values. e.g.
PixelSetColor( bgWand, '#0000ff' );
Gives me a blue background.

Unfortunately PixelSetColor( bgWand, 'none' ); gives me a black background and alpha of 255:

R:0 G:0 B:0 A:255

Version: 7.0.10.13
Language: Delphi

**Reply by xyz-coder** [reply]
Note: All the following give an opaque black background: "none", "transparent", "#0.......", "rgba(0, 0, 0, 0.0)"

**Comment by snibgo** [comment]
I have now tested this, and in a C program with MagickWand we need MagickSetBackgroundColor(), not MagickSetImageBackgroundColor().

Your program isn't C. I don't know what language it is.

**Comment by snibgo** [comment]
I don't know Delphi.

> PixelSetColor( bgWand, '#0000ff' );
Gives me a blue background.

Good.

With IM v7.0.8-64, for your input SVG and this C program:
```
#include <stdio.h>
#include <stdlib.h>
#include <MagickWand/MagickWand.h>

int main(int argc,char **argv)
{
#define ThrowWandException(wand) \
{ \
  char \
    *description; \
 \
  ExceptionType \
    severity; \
 \
  description=MagickGetException(wand,&severity); \
  (void) fprintf(stderr,"%s %s %lu %s\n",GetMagickModule(),description); \
  description=(char *) MagickRelinquishMemory(description); \
  exit(-1); \
}


  MagickWand *mwand = NewMagickWand();

// Set background to none
  PixelWand *bgWand = NewPixelWand();
  PixelSetColor( bgWand, "none" );

  MagickSetBackgroundColor( mwand, bgWand );

  if (!MagickReadImage(mwand, "lion-svgrepo-com.svg"))
    ThrowWandException(mwand);

// Check image has alpha
  int isAlpha = MagickGetImageAlphaChannel( mwand );
  if (!isAlpha)
    return(0);

  if (!MagickWriteImages (mwand, "out.png", MagickTrue))
    ThrowWandException(mwand);

  if (!MagickWriteImages (mwand, "txt:", MagickTrue))
    ThrowWandException(mwand);

  DestroyMagickWand( mwand );
  DestroyPixelWand( bgWand );

  return (0);
}
```
... the first few pixels are RGBA=(0,0,0,0), which is transparent black, as expected.

I suggest you try saving the image as a PNG file. Are the background pixels as expected, eg transparent when you have specified "none"?

**Reply by xyz-coder** [reply]
Thank you, your code does work for me. I'll work back to find the source of the issue in my code.
